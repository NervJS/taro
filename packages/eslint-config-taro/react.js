module.exports = Object.assign({}, require('./index'), {
  settings: {
    react: {
      pragma: 'React',
      version: 'detect'
    }
  }
})
