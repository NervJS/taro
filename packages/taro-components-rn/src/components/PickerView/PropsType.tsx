import React from 'react'

export type OnChangeEvent<T, E> = { detail: { value: T } & E }
export interface PickerViewProps<T, E = Record<string, unknown>> {
  children?: React.ReactNode;
  mode?: 'selector' | 'multiSelector' | 'time' | 'date' | 'region';
  value?: number[];
  onChange?: (event: OnChangeEvent<T, E>) => void;
  onCancel?: () => void;
  disabled?: boolean;
  itemStyle?: Record<string, unknown>;
  indicatorStyle?: Record<string, unknown>;
}
