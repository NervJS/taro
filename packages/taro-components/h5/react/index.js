import reactifyWc from './utils/reactify-wc'
import ReactInput from './components/input'

export const View = reactifyWc('taro-view-core')
export const Icon = reactifyWc('taro-icon-core')
export const Progress = reactifyWc('taro-progress-core')
export const RichText = reactifyWc('taro-rich-text-core')
export const Text = reactifyWc('taro-text-core')
export const Button = reactifyWc('taro-button-core')
export const Checkbox = reactifyWc('taro-checkbox-core')
export const CheckboxGroup = reactifyWc('taro-checkbox-group-core')
export const Editor = reactifyWc('taro-editor-core')
export const Form = reactifyWc('taro-form-core')
export const Input = ReactInput
export const Label = reactifyWc('taro-label-core')
export const Picker = reactifyWc('taro-picker-core')
export const PickerView = reactifyWc('taro-picker-view-core')
export const PickerViewColumn = reactifyWc('taro-picker-view-column-core')
export const Radio = reactifyWc('taro-radio-core')
export const RadioGroup = reactifyWc('taro-radio-group-core')
export const Slider = reactifyWc('taro-slider-core')
export const Switch = reactifyWc('taro-switch-core')
export const CoverImage = reactifyWc('taro-cover-image-core')
export const Textarea = reactifyWc('taro-textarea-core')
export const CoverView = reactifyWc('taro-cover-view-core')
export const MovableArea = reactifyWc('taro-movable-area-core')
export const MovableView = reactifyWc('taro-movable-view-core')
export const ScrollView = reactifyWc('taro-scroll-view-core')
export const Swiper = reactifyWc('taro-swiper-core')
export const SwiperItem = reactifyWc('taro-swiper-item-core')
export const FunctionalPageNavigator = reactifyWc('taro-functional-page-navigator-core')
export const Navigator = reactifyWc('taro-navigator-core')
export const Audio = reactifyWc('taro-audio-core')
export const Camera = reactifyWc('taro-camera-core')
export const Image = reactifyWc('taro-image-core')
export const LivePlayer = reactifyWc('taro-live-player-core')
export const Video = reactifyWc('taro-video-core')
export const Map = reactifyWc('taro-map-core')
export const Canvas = reactifyWc('taro-canvas-core')
export const Ad = reactifyWc('taro-ad-core')
export const OfficialAccount = reactifyWc('taro-official-account-core')
export const OpenData = reactifyWc('taro-open-data-core')
export const WebView = reactifyWc('taro-web-view-core')
export const NavigationBar = reactifyWc('taro-navigation-bar-core')
export const Block = reactifyWc('taro-block-core')
export const CustomWrapper = reactifyWc('taro-custom-wrapper-core')
