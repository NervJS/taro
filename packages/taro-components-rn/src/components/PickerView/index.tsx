import * as React from 'react'
import AntPickerView from '@ant-design/react-native/lib/picker-view'
import { noop } from '../../utils'

export default class _Picker extends React.Component<any> {
  static defaultProps = {
    data: [],
    value: []
  }

  onChange = (val: Record<string, any>): void => {
    const { onChange = noop } = this.props
    onChange({ detail: { value: val } })
  }

  handleChildren = (children: React.ReactChild[]): any[] => {
    return children.map((child: any, index: number) => {
      return {
        label: this.getLabelFromChildren(child),
        value: index
      }
    })
  }

  joinString = (data: string | any[] | React.ReactElement): string => {
    return (Array.isArray(data) ? data : [data]).join('')
  }

  getLabelFromChildren = (child: React.ReactElement): string => {
    return child.props && child.props.children ? this.getLabelFromChildren(child.props.children) : this.joinString(child)
  }

  getDataFromChildren = (children: React.ReactNode): any[] => {
    return (Array.isArray(children) ? children : [children]).map((child: any) => {
      return this.handleChildren(child.props && child.props.children ? child.props.children : [child])
    })
  }

  render(): JSX.Element | null {
    const { data, value, children, ...restProps } = this.props
    if (!children) return null
    return (
      <AntPickerView
        {...restProps}
        cols={1}
        value={value}
        data={data.length > 0 ? data : this.getDataFromChildren(children)}
        onChange={this.onChange}
        cascade={false}
      />
    )
  }
}
