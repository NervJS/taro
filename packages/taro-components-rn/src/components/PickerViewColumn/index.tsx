
import * as React from 'react'
import View from '../View'

export default class _PickerViewColumn extends React.Component<any> {
  static defaultProps = {
    mode: 'selector',
  }

  render (): JSX.Element {
    return (
      <View {...this.props} />
    )
  }
}
