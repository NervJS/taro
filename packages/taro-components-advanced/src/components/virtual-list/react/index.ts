import { ScrollView, View } from '@tarojs/components'
import React from 'react'

import List from './list'
import outerWrapper from './wrapper'

import type { VirtualListProps } from '../'
import type { IProps } from '../preset'

const VirtualList = React.forwardRef(function VirtualList (props: VirtualListProps, ref) {
  const {
    direction = 'ltr',
    outerElementType = ScrollView,
    innerElementType = View,
    itemElementType = View,
    initialScrollOffset = 0,
    overscanCount = 1,
    ...rest
  } = props as IProps

  if ('children' in rest) {
    console.warn('Taro(VirtualList): children props have been deprecated. ' + 'Please use the item props instead.')
    rest.item = rest.children as IProps['item']
  }
  if (rest.item instanceof Array) {
    console.warn('Taro(VirtualList): item should not be an array')
    rest.item = rest.item[0]
  }
  return React.createElement(List, {
    ref,
    ...rest,
    outerElementType,
    itemElementType,
    innerElementType,
    direction,
    initialScrollOffset,
    overscanCount,
    outerWrapper,
  })
})

export default VirtualList
