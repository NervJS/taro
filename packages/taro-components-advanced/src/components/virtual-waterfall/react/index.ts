import { ScrollView, View } from '@tarojs/components'
import React from 'react'

import Waterfall from './waterfall'
import outerWrapper from './wrapper'

import type { VirtualWaterfallProps } from '../'
import type { IProps } from '../preset'

const VirtualWaterfall = React.forwardRef(function VirtualWaterfall (props: VirtualWaterfallProps, ref) {
  const {
    outerElementType = ScrollView,
    innerElementType = View,
    itemElementType = View,
    initialScrollOffset = 0,
    overscanDistance = 50,
    ...rest
  } = props as IProps

  if ('children' in rest) {
    console.warn('Taro(VirtualWaterfall): children props have been deprecated. ' + 'Please use the item props instead.')
    rest.item = rest.children as IProps['item']
  }
  if (rest.item instanceof Array) {
    console.warn('Taro(VirtualWaterfall): item should not be an array')
    rest.item = rest.item[0]
  }
  return React.createElement(Waterfall, {
    ref,
    ...rest,
    outerElementType,
    itemElementType,
    innerElementType,
    initialScrollOffset,
    overscanDistance,
    outerWrapper,
  })
})

export default VirtualWaterfall
