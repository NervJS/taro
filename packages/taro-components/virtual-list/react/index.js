import React from 'react'
import FixedSizeList from './FixedSizeList'
import { ScrollView, View } from '@tarojs/components'

function convertPxToInt (style) {
  if (typeof style === 'string') {
    const str = style.toLowerCase()
    if (/px$/.test(str)) {
      return Number(str.replace(/px$/, ''))
    }
  }
  return style
}

const OuterScrollView = React.forwardRef(
  (props, ref) => {
    const { style, onScroll, onScrollNative, layout, ...rest } = props
    const handleScroll = event => {
      onScroll({
        ...event,
        currentTarget: {
          ...event.detail,
          clientWidth: convertPxToInt(style.width),
          clientHeight: convertPxToInt(style.height)
        }
      })

      if (typeof onScrollNative === 'function') {
        onScrollNative(event)
      }
    }

    return React.createElement(ScrollView, {
      ref,
      style,
      scrollY: layout === 'vertical',
      scrollX: layout === 'horizontal',
      onScroll: handleScroll,
      ...rest
    })
  }
)

const VirtualList = React.forwardRef((props, ref) => {
  const {
    direction = 'ltr',
    innerElementType = View,
    itemElementType = View,
    initialScrollOffset = 0,
    overscanCount = 1,
    ...rest
  } = props

  return React.createElement(FixedSizeList, {
    ref,
    ...rest,
    itemElementType,
    innerElementType,
    outerElementType: OuterScrollView,
    direction,
    initialScrollOffset,
    overscanCount
  })
})

export default VirtualList
