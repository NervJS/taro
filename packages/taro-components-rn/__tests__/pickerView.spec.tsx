// eslint-disable-next-line no-use-before-define
import * as React from 'react'
import { View } from 'react-native'
import renderer from 'react-test-renderer'
import PickerView from '../src/components/PickerView'
import PickerViewColumn from '../src/components/PickerViewColumn'

describe('PickerView', () => {
  it('PickerView render', () => {
    const years: number[] = []
    const months: number[] = []
    const days: number[] = []
    for (let i = 1990; i <= 2020; i++) {
      years.push(i)
    }
    for (let i = 1; i <= 12; i++) {
      months.push(i)
    }
    for (let i = 1; i <= 31; i++) {
      days.push(i)
    }
    const component = renderer.create(
      <PickerView indicatorStyle='height: 50px;' style='width: 100%; height: 300px;'>
        <PickerViewColumn>
          {years.map(item => {
            return (
              <View key={item}>{item}年</View>
            )
          })}
        </PickerViewColumn>
        <PickerViewColumn>
          {months.map(item => {
            return (
              <View key={item}>{item}月</View>
            )
          })}
        </PickerViewColumn>
        <PickerViewColumn>
          {days.map(item => {
            return (
              <View key={item}>{item}日</View>
            )
          })}
        </PickerViewColumn>
      </PickerView>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
