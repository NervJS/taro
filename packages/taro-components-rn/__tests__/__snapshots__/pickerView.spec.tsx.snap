// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerView PickerView render 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCPicker
      items={
        Array [
          Object {
            "label": "1990年",
            "textColor": undefined,
            "value": 0,
          },
          Object {
            "label": "1991年",
            "textColor": undefined,
            "value": 1,
          },
          Object {
            "label": "1992年",
            "textColor": undefined,
            "value": 2,
          },
          Object {
            "label": "1993年",
            "textColor": undefined,
            "value": 3,
          },
          Object {
            "label": "1994年",
            "textColor": undefined,
            "value": 4,
          },
          Object {
            "label": "1995年",
            "textColor": undefined,
            "value": 5,
          },
          Object {
            "label": "1996年",
            "textColor": undefined,
            "value": 6,
          },
          Object {
            "label": "1997年",
            "textColor": undefined,
            "value": 7,
          },
          Object {
            "label": "1998年",
            "textColor": undefined,
            "value": 8,
          },
          Object {
            "label": "1999年",
            "textColor": undefined,
            "value": 9,
          },
          Object {
            "label": "2000年",
            "textColor": undefined,
            "value": 10,
          },
          Object {
            "label": "2001年",
            "textColor": undefined,
            "value": 11,
          },
          Object {
            "label": "2002年",
            "textColor": undefined,
            "value": 12,
          },
          Object {
            "label": "2003年",
            "textColor": undefined,
            "value": 13,
          },
          Object {
            "label": "2004年",
            "textColor": undefined,
            "value": 14,
          },
          Object {
            "label": "2005年",
            "textColor": undefined,
            "value": 15,
          },
          Object {
            "label": "2006年",
            "textColor": undefined,
            "value": 16,
          },
          Object {
            "label": "2007年",
            "textColor": undefined,
            "value": 17,
          },
          Object {
            "label": "2008年",
            "textColor": undefined,
            "value": 18,
          },
          Object {
            "label": "2009年",
            "textColor": undefined,
            "value": 19,
          },
          Object {
            "label": "2010年",
            "textColor": undefined,
            "value": 20,
          },
          Object {
            "label": "2011年",
            "textColor": undefined,
            "value": 21,
          },
          Object {
            "label": "2012年",
            "textColor": undefined,
            "value": 22,
          },
          Object {
            "label": "2013年",
            "textColor": undefined,
            "value": 23,
          },
          Object {
            "label": "2014年",
            "textColor": undefined,
            "value": 24,
          },
          Object {
            "label": "2015年",
            "textColor": undefined,
            "value": 25,
          },
          Object {
            "label": "2016年",
            "textColor": undefined,
            "value": 26,
          },
          Object {
            "label": "2017年",
            "textColor": undefined,
            "value": 27,
          },
          Object {
            "label": "2018年",
            "textColor": undefined,
            "value": 28,
          },
          Object {
            "label": "2019年",
            "textColor": undefined,
            "value": 29,
          },
          Object {
            "label": "2020年",
            "textColor": undefined,
            "value": 30,
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCPicker
      items={
        Array [
          Object {
            "label": "1月",
            "textColor": undefined,
            "value": 0,
          },
          Object {
            "label": "2月",
            "textColor": undefined,
            "value": 1,
          },
          Object {
            "label": "3月",
            "textColor": undefined,
            "value": 2,
          },
          Object {
            "label": "4月",
            "textColor": undefined,
            "value": 3,
          },
          Object {
            "label": "5月",
            "textColor": undefined,
            "value": 4,
          },
          Object {
            "label": "6月",
            "textColor": undefined,
            "value": 5,
          },
          Object {
            "label": "7月",
            "textColor": undefined,
            "value": 6,
          },
          Object {
            "label": "8月",
            "textColor": undefined,
            "value": 7,
          },
          Object {
            "label": "9月",
            "textColor": undefined,
            "value": 8,
          },
          Object {
            "label": "10月",
            "textColor": undefined,
            "value": 9,
          },
          Object {
            "label": "11月",
            "textColor": undefined,
            "value": 10,
          },
          Object {
            "label": "12月",
            "textColor": undefined,
            "value": 11,
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCPicker
      items={
        Array [
          Object {
            "label": "1日",
            "textColor": undefined,
            "value": 0,
          },
          Object {
            "label": "2日",
            "textColor": undefined,
            "value": 1,
          },
          Object {
            "label": "3日",
            "textColor": undefined,
            "value": 2,
          },
          Object {
            "label": "4日",
            "textColor": undefined,
            "value": 3,
          },
          Object {
            "label": "5日",
            "textColor": undefined,
            "value": 4,
          },
          Object {
            "label": "6日",
            "textColor": undefined,
            "value": 5,
          },
          Object {
            "label": "7日",
            "textColor": undefined,
            "value": 6,
          },
          Object {
            "label": "8日",
            "textColor": undefined,
            "value": 7,
          },
          Object {
            "label": "9日",
            "textColor": undefined,
            "value": 8,
          },
          Object {
            "label": "10日",
            "textColor": undefined,
            "value": 9,
          },
          Object {
            "label": "11日",
            "textColor": undefined,
            "value": 10,
          },
          Object {
            "label": "12日",
            "textColor": undefined,
            "value": 11,
          },
          Object {
            "label": "13日",
            "textColor": undefined,
            "value": 12,
          },
          Object {
            "label": "14日",
            "textColor": undefined,
            "value": 13,
          },
          Object {
            "label": "15日",
            "textColor": undefined,
            "value": 14,
          },
          Object {
            "label": "16日",
            "textColor": undefined,
            "value": 15,
          },
          Object {
            "label": "17日",
            "textColor": undefined,
            "value": 16,
          },
          Object {
            "label": "18日",
            "textColor": undefined,
            "value": 17,
          },
          Object {
            "label": "19日",
            "textColor": undefined,
            "value": 18,
          },
          Object {
            "label": "20日",
            "textColor": undefined,
            "value": 19,
          },
          Object {
            "label": "21日",
            "textColor": undefined,
            "value": 20,
          },
          Object {
            "label": "22日",
            "textColor": undefined,
            "value": 21,
          },
          Object {
            "label": "23日",
            "textColor": undefined,
            "value": 22,
          },
          Object {
            "label": "24日",
            "textColor": undefined,
            "value": 23,
          },
          Object {
            "label": "25日",
            "textColor": undefined,
            "value": 24,
          },
          Object {
            "label": "26日",
            "textColor": undefined,
            "value": 25,
          },
          Object {
            "label": "27日",
            "textColor": undefined,
            "value": 26,
          },
          Object {
            "label": "28日",
            "textColor": undefined,
            "value": 27,
          },
          Object {
            "label": "29日",
            "textColor": undefined,
            "value": 28,
          },
          Object {
            "label": "30日",
            "textColor": undefined,
            "value": 29,
          },
          Object {
            "label": "31日",
            "textColor": undefined,
            "value": 30,
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
